### ISSUE TITLE
Using 'travis encrypt "key" --add' with jobs/stages and multiple deploy providers

### ISSUE THREAD

**Issue by Chadsr** [travis-ci/travis-ci#9082]
Hi,

This possibly has a very simple solution, but after some hours of searching I have not found one. It's also possible that I have gravely misunderstood some of the syntax I am attempting.

I have a rather complex .travis.yml which I am trying to add encrypted keys to, using the cli tool like so:
`travis encrypt "some_key" --add some.path`

My problem arises when I have a yml with many key/value pairs such as:

```
jobs:
  include:
    stage: some_stage
      deploy:
        - provider: provider1
          key:
        - provider: provider2
          key:
```

So in this mock-example, I am attempting to add an encrypted secret to each of the providers. I'm unable to find a way to reference the provider value (e.g. provider1) so that travis encrypt will insert the secret like so:

```
jobs:
  include:
    stage: some_stage
      deploy:
        - provider: provider1
          key:
            secure: some_encrypted_secret
        - provider: provider2
          key:
            secure: some_other_encrypted_secret
```

So, in summary i'd like to ask if what I am attempting is possible and if so, how exactly can I add a secure key to a provider with `--add` in a structure such as above?

Thanks for your time.

**Comment by BanzaiMan** [comment]
I am sorry to say, but CLI client's `--add` option is rather simplistic and unable to handle recent features such as build stages.

**Comment by Chadsr** [comment]
Good to know (and thanks for the super fast reply).
I was not sure if I had missed some syntax but it seems Google had no solution, because there isn't one yet.

This leads me to one last query before I close the issue.
Am I correct in thinking that the secure key's location is hashed into the key? I have been under the impression that an encrypted key changes depending on its intended location in the config, which is why I have been avoiding encrypting via stdin with just `travis encrypt`. 

If my assumption is in fact false, then I will return to encrypting via the above method.

Thanks again.

**Comment by BanzaiMan** [comment]
https://docs.travis-ci.com/user/encryption-keys#Detailed-Discussion has how `secure` key works.

**Comment by Chadsr** [comment]
It looks like I have missed some important information on that topic, so thank you for providing the link. I must have skimmed over it after discovering that `--add` could solve my initial problems with ease.

I will close the topic now, as I imagine extending the `--add` argument is already on the roadmap, when build stages and such have matured.
